### ISSUE TITLE
pyodbc.OperationalError: ('08001', '[08001] [Microsoft][ODBC Driver 17 for SQL Server]TCP Provider: Error code 0x2746 (10054) (SQLDriverConnect)')

### ISSUE THREAD

**Issue by krushika29** [mkleehammer/pyodbc#853]
### Environment

- Python: 3.8.5 (64-bit)
- pyodbc: 4.0.30
- OS: Ubuntu 20.04.1 LTS
- DB: SQL Server 13.0.5026.0
- driver: ODBC Driver 17 for SQL Server

### Issue

**pyodbc.OperationalError: ('08001', '[08001] [Microsoft][ODBC Driver 17 for SQL Server]TCP Provider: Error code 0x2746 (10054) (SQLDriverConnect)')**

### Additonal Information
Installed ODBC Driver following the steps mentioned here for Ubuntu 20.04 [Click Here](https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver15#ubuntu17)

Trying to connect to a SQL Server database from a Python web application using Flask.
Here is the python code that connects with the database - 

```
from flask import Flask, render_template, request, redirect, make_response
import pyodbc, datetime
import pandas as pd

app = Flask(__name__)


#connect_db function is used to connect with the database for read and write purposes
def connect_db():
    connection = pyodbc.connect(r'Driver={ODBC Driver 17 for SQL Server};Server=servername;Database=databasename;Trusted_Connection=yes',autocommit=True)
    return connection


#index function calls the homepage
@app.route('/')
def index():
    con = connect_db()
    cursor = con.cursor()
    return render_template('index.html')
```

The following is the output for checking pyodbc drivers - 
`
>>> import pyodbc
>>> pyodbc.drivers()
['ODBC Driver 17 for SQL Server']
>>>

`
I have all the drivers and packages installed correctly, and this is the third time I am installing them and I have the same error. Please help

**Comment by krushika29** [comment]
I have already tried this 
[Click here](https://github.com/microsoft/msphpsql/issues/1112#issuecomment-609972220)

This doesn't work for me either

**Comment by v-chojas** [comment]
Can you try to connect with the ODBC driver directly, using isql from the command line?

`isql -v -k '<connection string here>'`

**Comment by krushika29** [comment]
@v-chojas I have implemented the solution mentioned [here](https://github.com/microsoft/msphpsql/issues/1112#issuecomment-609972220)

After a quick restart of the system, it worked!
Do you know why this issue, is it because I am trying to access an older version of SQL Server database i.e. 2016 from ODBC 17?

**Comment by v-chojas** [comment]
TLS version mismatch. It is not directly the driver, but the OpenSSL it uses which default to TLS 1.2+ only for the later versions.

**Comment by cbare** [comment]
Just a note for those that end up here trying to solve a similar issue: I saw the same error from `pyodbc.connect` running in Windows Subsystem for Linux (5.10.16.3-microsoft-standard-WSL2) / Ubuntu 20.04.3, and OpenSSL 1.1.1k  25 Mar 2021. When connected to VPN I got the error. Error disappeared after disconnecting. The same code and connect string worked over VPN when running from Powershell. I guess some interaction between OpenSSL and our VPN is at play?
